CenterForOpenScience/osf.io
# [Hotfix] Fix amp in dashboard and po (take 3) [OSF-6853]

## Purpose
& become &amp; on the Dashboard and Project Organizer.
The ideal solution to this would handle amps better on the backend. This is a cosmetic front end hack/fix to make amps show up properly. Where possible I caught the titles early, but a lot of the time.



## Changes
Replace &amp; with & on PO and Dashboard

## Side effects

This is a hack, so manually making the name contain &amp; will convert it to & in these displays.

## Ticket
[#OSF-6853]
https://openscience.atlassian.net/browse/OSF-6853

## QA Considerations
Test a project name that contains &
On the Dashboard project widget:
1. Make sure titles in the dashboard handle amps
2. Make sure each level of parent-grandparent handles amps (picture)
![screen shot 2016-11-01 at 10 26 54 am](https://cloud.githubusercontent.com/assets/1322421/19892963/bca238b8-a01d-11e6-9f8a-4caae4ab3263.png)
3. Make sure searching matches amp properly (e.g., title = cat&dog, quick search cat&d matches; cat&amp;dog will still match as well iirc)
4. Make sure projects loaded by pressing the 'more' down arrow handle amps

Project Organizer:
1. Make sure titles handle amps
2. Make sure child nodes handle amps
3. Make sure the side bar (information/activity area) handles amps (NOTE: Node logs in Activity do not handle amps)
4. Make sure newly created collections handle amps
5. Make sure collections handle amps on refresh (e.g., previously created ones, not new ones)
6. Make sure renaming collections properly populates the amp
7. Make sure renamed collections handle the amp
8. Check that dragging a project into a collection, the text being dragged handles the amps
9. Check that filtering filters on & not &amp;
10. Make sure breadcrumbs (both parent and child) handle amps
11. Make sure component with private parent (no access) does not break dashboard
I think that is all of the places this changed.
